corpan/dja/cor/models.py

```diff
@@ -56,3 +59,102 @@ def __str__(self):
         if self.romanization:
             return f"[{self.language.code}] {self.text} ({self.romanization})"
         return f"[{self.language.code}] {self.text}"
+
+
+class Narrator(models.Model):
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    name = models.CharField(max_length=100)
+    language = models.ForeignKey("Language", on_delete=models.PROTECT)
+    description_pack = models.ForeignKey(
+        "Pack",
+        on_delete=models.SET_NULL,
+        null=True,
+        blank=True,
+        related_name="narrator_descriptions",
+        help_text="Pack whose entries describe this narrator. Blank title = internal use only.",
+    )
+
+    def __str__(self):
+        return self.name
+
+    def description_text(self, language):
+        """Return description text in the requested language, joined as paragraphs."""
+        if not self.description_pack:
+            return ""
+        return self.description_pack.get_full_text(language)
+
+
+class Pack(models.Model):
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    title = models.CharField(max_length=200, blank=True)
+    narrator = models.ForeignKey(
+        Narrator, on_delete=models.PROTECT, null=True, blank=True, related_name="packs"
+    )
+    description_pack = models.ForeignKey(
+        "self",
+        on_delete=models.SET_NULL,
+        null=True,
+        blank=True,
+        related_name="describes_packs",
+        help_text="Optional pack whose entries form a translatable description for this pack.",
+    )
+
+    def __str__(self):
+        return self.title or f"Pack {self.id}"
+
+    # --- Helper methods ---
+
+    @property
+    def is_public(self):
+        """Public packs have a title."""
+        return bool(self.title.strip())
+
+    @classmethod
+    def public(cls):
+        """Queryset of all packs with titles."""
+        return cls.objects.filter(~Q(title=""), ~Q(title=None))
+
+    def get_full_text(self, language):
+        """Return concatenated text of all entries in this pack in the given language."""
+        texts = []
+        for pe in self.entries.select_related("entry"):
+            if translation := pe.entry.translations.filter(language=language).first():
+                texts.append(translation.text)
+            else:
+                texts.append(pe.entry.en_text)  # fallback to English
+        return " ".join(texts)
+
+    @classmethod
+    def create_from_text(
+        cls, text, language, narrator=None, title="", llm_provider="openai"
+    ):
+        """
+        Create a pack by splitting description into sentences (naive split on '.').
```

Comment:
comment is wrong - this should be a robust split on all possible breaks in all scripts not "naive split on '.'"

Reply:
this method was where we started but it's not actually used and should just be removed in favor of `cor.packs.service` I suppose.